scalding-beam/src/test/scala/com/twitter/scalding/beam_backend/BeamBackendTests.scala

```diff
@@ -136,6 +136,23 @@ class BeamBackendTests extends FunSuite with BeforeAndAfter {
     )
   }
 
+  test("HashJoin"){
+    beamMatchesSeq({
+      val smaller = TypedPipe.from(Seq((0, 0), (2, 1)))
+      TypedPipe.from(Seq((0, 1), (0, 2), (1, 0))).hashJoin(smaller)
+    },
+      Seq((0, (1, 0)), (0, (2, 0)))
+    )
+  }
+
+  test("CoGroup"){
+    beamMatchesSeq({
+      val tp1: TypedPipe[(Int, Int)] = TypedPipe.from(Seq((1, 2), (1, 3)))
+      val tp2: TypedPipe[(Int, Int)] = TypedPipe.from(Seq((1, 7)))
```

Comment:
can we add tests with multiple values per keys on left, right, both... and also filtering: where some keys only appear on the right or left.

Also, can we test join, leftJoin, rightJoin, outerJoin since those are the most common join functions?